protocol/parser/parser.go

```diff
@@ -108,10 +109,10 @@ func parseInputFromParamsWithGenericParsers(rpcInput RPCInput, genericParsers []
 	}
 
 	parsed := NewParsedInput()
-	parsedBlock := genericParserResult.GetBlock()
-	if parsedBlock != spectypes.NOT_APPLICABLE {
+	parsedBlock := genericParserResult.GetRawParsedData()
+	if parsedBlock != "" {
 		parsedSuccessfully = true
-		parsed.parsedBlock = parsedBlock
+		parsed.parsedDataRaw = parsedBlock
 	}
 
 	parsedBlockHashes, err := genericParserResult.GetBlockHashes()
```

Comment:
That's right. It doesn't matter if we parsed hashes or not, `parsedSuccessfully` will be true only if we parsed a raw block successfully.

Reply:
This is the current behavior, only because we still using the old block parser. Once we remove the old block parser, there will be no need for `parsedSuccessfully`

Reply:
lets rename it to parsedBlock

Reply:
Renamed in cb26380c328d45aad01aa15fc27a86f962876786

Reply:
to be clear: add the parsed successfully to the hashes flow, i don't care if in our specific case right now the caller doesnt have to have it for now or is aware what is going on inside, currently this is just bad practice and this code will be relevant in the future

Reply:
Fixed in 95e832b11a96e03393e9f4bef9c96d143617b3f9